### PR TITLE
Fix build projects

### PR DESCRIPTION
**Pull request by boeckMt**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The Project is [building and passes all tests](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/master/DEVELOPMENT.md#further-you-can-test-and-build-locally) without errors

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The build of the github actions is **not** failing for **missing dependencies** (projects which depends on other projects).

For example since `@dlr-eoc/utils-maps` is used in `@dlr-eoc/map-ol` runing the dependency check before the build, show's the error (which is fixed in d83bdafb092fad4cffb3b0749dcfd9332fad8a29)

```
  project:  map-ol
  projectPath:  /home/runner/work/ukis-frontend-libraries/ukis-frontend-libraries/projects/map-ol
  missingDependencies: {
	"@dlr-eoc/utils-maps": [
		"/home/runner/work/ukis-frontend-libraries/ukis-frontend-libraries/projects/map-ol/src/lib/map-ol.service.ts"
	]
  }
```
but the build is passing fine and nobody recognises about it.

Missing dependencies change the dependency graph and terfore the build order of the projects could change. 
Because the build relies on a correct order (otherwise the can be errors like `'rootDir' is expected to contain all source files`) this should not happen.

Furthermore the use of `NG.default` to run angular commands sometimes fails because it triggers the next command before the last is completely finished.

Issue Number: N/A

## What is the new behavior?

- The script to build the libraries now returns `process.exit(1)` if there are missing dependencies.
- Each `NG.default` command now runs in its own process.
- The creation of the `dependency graph` and the `nodes` and `edges` for the `toposort` where refactored.
- A few type errors have been fixed.
- Setup node action and there node version (in the github workflows) are updated. (preparation for )

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Is it part of one/more [packages](https://github.com/dlr-eoc/ukis-frontend-libraries/packages) and which?

- @dlr-eoc/map-ol
- @dlr-eoc/utils-maps
- demo-maps
- scripts/library

### CODE REVIEW

**Comment by boeckMt** [comment]
@MichaelLangbein and @voinSR after this PR and  #59 I like to create a new version `v7.3.0`

**Comment by MichaelLangbein** [comment]
> @MichaelLangbein and @voinSR after this PR and #59 I like to create a new version `v7.3.0`

Sounds good to me!

File: projects/demo-maps/src/app/route-components/route-example-olperformance/services/largelayers.service.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ import {get as getProjection, getTransform} from 'ol/proj';
 import {register} from 'ol/proj/proj4';
 import proj4 from 'proj4';
 import { HttpClient } from '@angular/common/http';
-import { LayersService, VectorLayer, CustomLayer } from '@dlr-eoc/services-layers/src/public-api';
-import { MapOlService } from '@dlr-eoc/map-ol/src/public-api';
+import { LayersService, VectorLayer, CustomLayer } from '@dlr-eoc/services-layers';
+import { MapOlService } from '@dlr-eoc/map-ol';
```
Comment (MichaelLangbein) [inline projects/demo-maps/src/app/route-components/route-example-olperformance/services/largelayers.service.ts:16]: That `src/public-api` should not have been here to begin with - thanks a lot for removing it!

File: projects/utils-maps/src/lib/ol/utils-ol.ts [hunk 2]
```diff
@@ -75,7 +75,11 @@ export function mapToSingleCanvas(map: Map, targetCanvas: HTMLCanvasElement | Of
                 // Step 3: copy source bitmap to target-canvas.
                 targetContext.drawImage(sourceCanvas, 0, 0, sourceCanvas.clientWidth, sourceCanvas.clientHeight, 0, 0, targetCanvas.width, targetCanvas.height);
             });
-            subscriptions.push(key);
+            if (Array.isArray(key)){
```
Comment (MichaelLangbein) [inline projects/utils-maps/src/lib/ol/utils-ol.ts:78]: >> `if (Array.isArray(key)){`

Wasn't aware that a subscription-key can be an array. Thanks for catching that. Did this cause any issues anywhere?

**Review by MichaelLangbein — APPROVED** [review]
Changes perfectly fine with me.
Also: I realize that this CI work is not easy. Much appreciated!

File: projects/utils-maps/src/lib/ol/utils-ol.ts (hunk 2 above)
Comment (boeckMt) [reply projects/utils-maps/src/lib/ol/utils-ol.ts:78]: The typescript compiler made me aware of this :)
I don't think that it made any issues, but compiler errors should be fixed.

**Pull request merged**